### ISSUE TITLE
`empty_enum` fires on enums that are only empty because of `#[cfg]`

### ISSUE THREAD

**Issue by ada4a** [rust-lang/rust-clippy#15910]
### Summary

The lint fires on enums which happen to have all their variants `cfg`-d out.

This is called out in the lint description as a known problem:
https://github.com/rust-lang/rust-clippy/blob/1ecb18a18211c86a2a2f00ef3b0ef4c54f54d0d2/clippy_lints/src/empty_enum.rs#L28-L30

### Lint Name

empty_enum

### Reproducer

I tried this code:

```rust
#![allow(unused)]
#![warn(clippy::empty_enum)]

enum NotReallyEmpty {
    #[cfg(false)]
    PhantomVariant,
}
```

I saw this happen:

```
error: enum with no variants
  --> tests/ui/empty_enum.rs:8:1
   |
LL | / enum EmptyWithCfg {
LL | |     #[cfg(false)]
LL | |     PhantomVariant,
LL | | }
   | |_^
   |
   = help: consider using the uninhabited type `!` (never type) or a wrapper around it to introduce a type which can't be instantiated
```

I expected to see this happen:
no warning

### Version

```text
rustc 1.90.0 (1159e78c4 2025-09-14)
binary: rustc
commit-hash: 1159e78c4747b02ef996e55082b704c09b970588
commit-date: 2025-09-14
host: x86_64-unknown-linux-gnu
release: 1.90.0
LLVM version: 20.1.8
```

### Additional Labels

@rustbot claim
